lib/graphql/client/schema/object_type.rb

```diff
@@ -58,6 +61,9 @@ def initialize(klass, defined_fields, definition, spreads)
               name = ActiveSupport::Inflector.underscore(attr)
               @klass::READERS[:"#{name}"] ||= attr
               @klass::PREDICATES[:"#{name}?"] ||= attr
+
+              @klass.undef_method(name) if BASE_RUBY_METHODS.include?(name) && @klass.method_defined?(name)
+              @klass.undef_method("#{name}?") if BASE_RUBY_METHODS.include?("#{name}?") && @klass.method_defined?("#{name}?")
```

Comment:
Are there other precedents for doing this in GraphQL Ruby resolvers? My concern is that this produces some really opaque side effects. Definitely feels like this would be at least a minor version bump on the gem; major to be safe.